### PR TITLE
fix(Workspace): adds pipeline user check to fetch pipelines

### PR DESCRIPTION
**Pull request by nazarfil**
workaround for PipelineUser permissions to fetch workspace

### CODE REVIEW

File: hexa/workspaces/schema/queries.py [hunk 1]
```diff
@@ -25,7 +26,13 @@ def resolve_workspaces(_, info, query=None, page=1, perPage=15):
 def resolve_workspace(_, info, **kwargs):
     request = info.context["request"]
     try:
-        return Workspace.objects.filter_for_user(request.user).get(slug=kwargs["slug"])
+        # TODO: add pipeline user permission check
```
Comment (qgerome) [inline hexa/workspaces/schema/queries.py:29]: Why is this needed since we already check this in the `filter_for_user` method of the Queryset ?

Comment (qgerome) [reply hexa/workspaces/schema/queries.py:29]: I was referencing the body of your fix that is already implemented in Workspace.filter_for_user.

Comment (nazarfil) [reply hexa/workspaces/schema/queries.py:29]: It was not working.
Similar to what was added to the workspace `workspace.filter_for_user` for `createDataset` mutation. 
If it is expected from  the check inside `filter_for_user` to perform this for PipelineUser, then I think it is worth searching for the reason why it is not working in here
